lib/pbench/test/unit/agent/task/test_tools_kill.py

```diff
@@ -0,0 +1,440 @@
+                self._do = do
+
+            def children(self, recursive: Optional[bool] = False):
+                assert recursive
+                if not self._children:
+                    return
+                for child in self._children:
+                    yield child
+
+            def cmdline(self):
+                return [
+                    "arg0",
+                    "arg1",
+                    "arg2" if not self._uuid else self._uuid,
+                    "arg3",
+                ]
+
+            def kill(self):
+                if self._do == "nosuch":
+                    action_list.append(("kill-nosuch", self.pid, self._uuid))
+                    raise psutil.NoSuchProcess(self.pid)
+                if self._do == "exc":
+                    action_list.append(("kill-except", self.pid, self._uuid))
+                    raise Exception("generic exception")
+                action_list.append(("kill", self.pid, self._uuid))
+
+        class mock_psutil:
+            """Even though the target `psutil` is a module, we just use a
+            class with a static method to duck-type it.
+            """
+
+            @staticmethod
+            def process_iter():
+                # First pid has 3 children
+                yield MockProcess(
+                    12345,
+                    uuid=uuid_list[0],
+                    children=[
+                        MockProcess(12346),
+                        MockProcess(12347),
+                        MockProcess(12348, do="nosuch"),
+                    ],
+                )
+                yield MockProcess(23456)
+                yield MockProcess(34567, uuid=uuid_list[1], do="exc")
+                yield MockProcess(45678, uuid=uuid_list[2], do="nosuch")
+
+        runner = CliRunner(mix_stderr=False)
+        monkeypatch.setattr("pbench.cli.agent.commands.tools.kill.psutil", mock_psutil)
+        result = runner.invoke(kill.main, uuid_list)
+        assert result.exit_code == 0
+        expected_action_list = [
+            ("kill", 12345, "uuid1abc"),
+            ("kill", 12346, None),
+            ("kill", 12347, None),
+            ("kill-nosuch", 12348, None),
+            ("kill-except", 34567, "uuid2def"),
+            ("kill-nosuch", 45678, "uuid3ghi"),
+        ]
+        assert action_list == expected_action_list, (
+            f"action_list={action_list!r},"
+            f" expected_action_list={expected_action_list!r},"
+            f" stdout={result.stdout!r},"
+            f" stderr={result.stdout!r}"
+        )
+
+    @staticmethod
+    def test_without_uuids_no_action(monkeypatch):
+        """Exercise the code path where no run directories are found, so there
+        is nothing to kill.
+        """
+
+        class MockPidSource:
+            """No-op mock for PidSource."""
+
+            def __init__(self, file_name: str, display_name: str):
+                self.file_name = file_name
+                self.display_name = display_name
```

Comment:
Do we need lines 298 & 299?  (Consider using [unittest.mock.Mock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock) instead of rolling your own mock-class.)